libmuscle/cpp/build/libmuscle/Makefile

```diff
@@ -31,8 +31,16 @@ endif
 
 tests_deps := libmuscle_d.a version.h
 
+# Checking if we are installing and have MPI libs
+INSTALL_LIBS := $(findstring install,$(MAKECMDGOALS))
+LIBMUSCLE_MPI_SO_PRESENT := $(wildcard libmuscle_mpi.so)
+INSTALLING_WITH_MPI_LIBS := $(and $(INSTALL_LIBS),$(LIBMUSCLE_MPI_SO_PRESENT))
+
 # if we're compiling with MPI, or we are installing and have MPI libs
-ifeq ($(if $(or $(MUSCLE_ENABLE_CPP_MPI),$(and $(findstring install,$(MAKECMDGOALS)),$(wildcard libmuscle_mpi.so))),1,0),1)
+USE_MPI := $(or $(MUSCLE_ENABLE_CPP_MPI),$(INSTALLING_WITH_MPI_LIBS))
+MAKE_USE_MPI_FLAG := $(if $(USE_MPI),1,0)
```

Comment:
I'd prefer just `USE_MPI_FLAG` here, the `MAKE_` prefix is a bit confusing to me.